eng/common/scripts/job-matrix/job-matrix-functions.ps1

```diff
@@ -490,7 +490,7 @@ function CloneOrderedDictionary([System.Collections.Specialized.OrderedDictionar
 function SerializePipelineMatrix([Array]$matrix) {
     $pipelineMatrix = [Ordered]@{}
     foreach ($entry in $matrix) {
-        if ($pipelineMatrix.Contains($entry.Name)) {
+        if ($pipelineMatrix.Contains($entry.name)) {
```

Comment:
Bug! I managed to stumble upon this by utilizing an OrderedHashTable on the input, which caused `name` vs `Name` to blow up. I figured out

> oh Ben _never_ intended this to be capital `Name`, it was just working because the type may have been forgiving on capitalization before.

I know this because _in the error output_ you call it lower-case `name`.

Reply:
Yes. Confirmed. HashTables are case insensitive. OrderedHashTables are not. I am passing in a MatrixResult cloned, which turned it into an OrderedHashTable from a HashTable.

I tend to think this should be accessing the right property name anyway, so I should keep it, but I think I can account for this by converting the cloned result to a plain HashTable.